rust/processor/src/db/common/models/default_models/move_tables.rs

```diff
@@ -119,13 +130,55 @@ impl TableItem {
                 )
                 .unwrap(),
                 decoded_value: None,
-                last_transaction_version: transaction_version,
+                last_transaction_version: txn_version,
                 is_deleted: true,
             },
         )
     }
 }
 
+impl TableItemConvertible for TableItem {
+    fn from_raw(raw_item: &RawTableItem) -> Self {
+        TableItem {
+            transaction_version: raw_item.txn_version,
+            write_set_change_index: raw_item.write_set_change_index,
+            transaction_block_height: raw_item.transaction_block_height,
+            key: raw_item.table_key.clone(),
+            table_handle: raw_item.table_handle.clone(),
+            decoded_key: serde_json::from_str(raw_item.decoded_key.as_str()).unwrap(),
+            decoded_value: raw_item
+                .decoded_value
+                .clone()
+                .map(|v| serde_json::from_str(v.as_str()).unwrap()),
+            is_deleted: raw_item.is_deleted,
+        }
+    }
+}
+
+impl TableItem {
+    pub fn from_write_table_item(
+        write_table_item: &WriteTableItem,
+        write_set_change_index: i64,
+        transaction_version: i64,
+        transaction_block_height: i64,
+    ) -> Self {
+        Self {
+            transaction_version,
+            write_set_change_index,
+            transaction_block_height,
+            key: write_table_item.key.to_string(),
+            table_handle: standardize_address(&write_table_item.handle.to_string()),
+            decoded_key: serde_json::from_str(write_table_item.data.as_ref().unwrap().key.as_str())
+                .unwrap(),
+            decoded_value: serde_json::from_str(
+                write_table_item.data.as_ref().unwrap().value.as_str(),
+            )
+            .unwrap(),
+            is_deleted: false,
+        }
+    }
```

Comment:
why does this need to be defined again? this looks the same as `RawTableItem::from_write_table_item`

Reply:
I updated the original TableItem::from_write_table_item to return only TableItem bc current_table_item is not used like in `delegator_balances.rs`. not sure what's better mark unused with underscore or add this. 
we could move this RawTableItem and reuse too. wdyt?

Reply:
I think underscore to represent unused models is fine tbh. Within each model, we could do the conversion too.